module-3/app/service/client/mysfitsTableClient.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+// Info is for logging
+var Info *log.Logger
+
+// Init initializes the logger
+func Init(
+	infoHandle io.Writer) {
+	Info = log.New(infoHandle,
+		"INFO: ",
+		log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+// Item is a value returned by query
+type Item struct {
+	MysfitId        string `json:"MysfitId"`
+	Name            string `json:"Name"`
+	Species         string `json:"Species"`
+	Description     string `json:"Description"`
+	Age             int    `json:"Age"`
+	GoodEvil        string `json:"GoodEvil"`
+	LawChaos        string `json:"LawChaos"`
+	ThumbImageUri   string `json:"ThumbImageUri"`
+	ProfileImageUri string `json:"ProfileImageUri"`
+	Likes           int    `json:"Likes"`
+	Adopted         bool   `json:"Adopted"`
+}
+
+// Items stores a list of Items
+type Items []Item
+
+// Get items as array of structs
+func getItems(items []map[string]*dynamodb.AttributeValue) Items {
+	var mysfitList Items
+
+	err := dynamodbattribute.UnmarshalListOfMaps(items, &mysfitList)
+	if err != nil {
+		println("Got error unmarshalling items:")
+		println(err.Error())
+		return nil
+	}
+
+	return mysfitList
+}
+
+// getJSONStringFromItems creates a JSON string from the items from a scan or query
+func getJSONStringFromItems(items []map[string]*dynamodb.AttributeValue) string {
+	output := "{\"mysfits\": ["
+	myItems := getItems(items)
+
+	len := len(myItems)
+
+	for i, item := range myItems {
+		iJSON, err := json.Marshal(item)
```

Comment:
Is it possible to get the raw json list of Mysfits from Dynamo and return that directly instead of first unmarshalling on line 50 and then re-marshalling here?

Reply:
I tried but it gave me a bunch of nulls.